gwt-localizable-processor/src/main/java/org/gwtproject/i18n/processor/LocalizableProcessingStep.java

```diff
@@ -47,105 +58,380 @@ public LocalizableProcessingStep(ProcessingEnvironment processingEnv) {
+          ParameterizedTypeName.get(Map.class, String.class, String.class),
+          key);
+      builder
+          .beginControlFlow("if(args == null)")
+          .addStatement(
+              "args = new $T()",
+              ParameterizedTypeName.get(LinkedHashMap.class, String.class, String.class));
+
+      if (!property.isEmpty()) {
+        Arrays.asList(property.split("(?<![\\\\]),", -1))
+            .stream()
+            .map(String::trim)
+            .collect(Collectors.toSet())
+            .forEach(
+                valueKey -> {
+                  String valueProperty = localeResourceMap.get(locale).getProperty(valueKey);
+                  builder.addStatement("args.put($S, $S)", valueKey, valueProperty);
+                });
       }
 
-      // for each method that needs overriding, generate a method which
-      // either reads from the default annotation or just returns the
-      // method name itself, ignoring all args
-
-      // based on the types created and their locale,
-      // create a factor
-      TypeSpec factory =
-          TypeSpec.classBuilder(element.getSimpleName() + "_Factory")
-              .addMethod(
-                  MethodSpec.methodBuilder("create")
-                      .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
-                      .returns(ClassName.get(element.asType()))
-                      .addStatement("return create(System.getProperty(\"locale\", \"default\"))")
-                      .build())
-              .addMethod(
-                  MethodSpec.methodBuilder("create")
-                      .addParameter(String.class, "locale")
-                      .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
-                      .returns(ClassName.get(element.asType()))
-                      .addCode(makeSwitchCase(locales, keyToNameMapping))
-                      .build())
-              .build();
-      try {
-        JavaFile.builder(packageName, factory).build().writeTo(processingEnv.getFiler());
-      } catch (IOException e) {
-        e.printStackTrace();
+      builder.addStatement("args = $T.unmodifiableMap(args)", Collections.class);
+      builder.addStatement("cache.put($S, args)", key);
+      builder.endControlFlow();
+      builder.addStatement("return args");
+
+      return Optional.of(builder.build());
+    } else {
+      return Optional.empty();
+    }
+  }
+
+  private Optional<CodeBlock> getArrayMethodBody(
+      AptContext context,
+      ExecutableElement method,
+      String locale,
+      Map<String, Properties> localeResourceMap) {
+    CodeBlock.Builder builder = CodeBlock.builder();
+
+    String key = getKey(method);
+    String property = localeResourceMap.get(locale).getProperty(key);
+    if (nonNull(property)) {
+      builder.addStatement("$T[] args = ($T[])cache.get($S)", String.class, String.class, key);
+      builder.beginControlFlow("if(args == null)").beginControlFlow("$T[] writer=", String.class);
```

Comment:
I use this to inline initialize an array.

```java
if(args == null) {
      String[] writer= {
        "A",
        "B",
        "C",
        "D",
        "E",
        "F",
        "G",
      } ;
      cache.put("stringArrayABCDEFG", writer);
      return writer;
    }
```

Reply:
May want to be careful about that - as you noticed, it won't add the `;` for you, since code blocks don't need that. Might make more sense to chain .addCode or something?

Reply:
it does add the `;` when we do this 
`builder.endControlFlow("");`

Reply:
that's pretty gross :)